docker-compose.spark.yml

```diff
@@ -0,0 +1,61 @@
+version: '2'
+services:
+  spark-master:
+    image: spark-master
+    build:
+      context: ./worker-tasks
+      dockerfile: Dockerfile
+    image: quay.io/azavea/spark:1.6.2
+    restart: always
+    ports:
+      - "7077:7077"
+      - "8888:8888"
+    environment:
+      SPARK_LOCAL_DIRS: "/tmp"
+      SPARK_PUBLIC_DNS: "localhost"
+      SPARK_MASTER_WEBUI_PORT: "8888"
+      SPARK_DAEMON_MEMORY: "256m"
+      SPARK_DAEMON_JAVA_OPTS: "-Dspark.deploy.recoveryMode=FILESYSTEM -Dspark.deploy.recoveryDirectory=/spark-state"
+    volumes:
+      # Spark cluster state
+      - /spark-state
+      # Spark scratch space
+      - /tmp
+    entrypoint: spark-class
+    command: org.apache.spark.deploy.master.Master
+
+  spark-worker:
+    image: quay.io/azavea/spark:1.6.2
+    links:
+      - spark-master:spark.services.rf.internal
+    environment:
+      SPARK_LOCAL_DIRS: "/tmp"
+      SPARK_PUBLIC_DNS: "localhost"
+      SPARK_WORKER_WEBUI_PORT: "8889"
+      SPARK_WORKER_MEMORY: "512m"
+      SPARK_DAEMON_MEMORY: "512m"
+    volumes:
+      # Spark scratch space
+      - /tmp
+    entrypoint: spark-class
+    command:
+      - "org.apache.spark.deploy.worker.Worker"
+      - "spark://spark.services.rf.internal:7077"
+
+  spark-driver:
+    image: quay.io/azavea/spark:1.6.2
```

Comment:
My initial approach for this setup was to layout a base Spark development environment for Raster Foundry components that require it. Right now, the Spark setup consists of the OpenJDK JRE and Spark.

I haven't really taken a close look at GeoDocker until today (at least `geodocker-spark` and everything it inherits from), but there are a few aspects that seem to make it a bad fit for what we're trying to achieve right now:

- Based on CentOS and Oracle JDK
- Has multiple layers of base images pinned to `latest`
- Installs some components with Apache Bigtop (Hadoop, HDFS), some not (Spark)
- Seems setup to be used with GeoDocker as a whole vs. components by themselves

If we use GeoDocker to deploy GeoTrellis for Raster Foundry, then I think we'd definitely use it for development as well, but it isn't clear that that's the route we're going to take.